### DISCUSSION TITLE
Persistencia en caso de caida súbita de la red (Docker)

### DISCUSSION THREAD

**Discussion by rubensantibanezacosta** [opencanarias/taple-client#13] [Q&A]
En caso de que la red la tengamos levantada en contenedores, y hubiese una caida súbita de todos los nodos, entiendo que los datos existentes se perderían en el olvido... 

En el caso de querer persistir estos datos, o querer realizar backups, ¿en que rutas se almacenan los archivos?

En el caso de Docker, la idea sería crear volumenes en alguno de los nodos, a modo de aportar esa seguridad.

¿Teneis esto contemplado?

Saludos

**Comment by FlorentinPG** [comment]
La base de datos de datos que ahora mismo emplea TAPLE es LevelDB, que almacena todos sus datos en un directorio especificable por el usuario, de tal manera que un backup puede realizarse copiando dicho directorio en cualquier momento. Debido a que TAPLE ahora mismo se encuentra en desarrollo y está publicado para permitir a la comunidad hacer pruebas con él, el directorio por defecto en el que se crea la BBDD es /tmp/data, pero este puede ser modificado con la variable de entorno correspondiente (TAPLE_DATABASE_PATH). En lo que respecta a los volúmenes de Docker, es decisión del usuario decidir si quiere o no emplearlos para garantizar la permanencia de datos en los contenedores.

Saludos y gracias por la contribución.

**Comment by rubensantibanezacosta** [comment]
Muchas gracias. Lo pondré a prueba.
